src/main/java/at/hannibal2/skyhanni/features/mining/KingTalismanHelper.kt

```diff
@@ -200,19 +156,21 @@ object KingTalismanHelper {
     }
 
     private fun getKingTimes(): MutableMap<String, Long> {
-        val currentOffset = getCurrentOffset() ?: 0
+        val sbTime = SkyBlockTime.now()
+        val absDay = (sbTime.year - 1) * 372 + (sbTime.month - 1) * 31 + (sbTime.day - 1)
+        val kingIndex = (absDay + 4) % kingCircles.size
+        val circleStart = absDay - kingIndex
         val oneSBDay = 1000 * 60 * 20
         val oneCircleTime = oneSBDay * kingCircles.size
         val kingTime = mutableMapOf<String, Long>()
         for ((index, king) in kingCircles.withIndex()) {
-//             val startTime = SkyBlockTime(day = index + 2 - kingCircles.size)
-//             val startTime = SkyBlockTime(day = index - kingCircles.size)
-            val startTime = SkyBlockTime(day = index + currentOffset - kingCircles.size)
+            val startTime = SkyBlockTime.fromAbsoluteDay(index + circleStart)
+
             var timeNext = startTime.toMillis()
-            while (timeNext < System.currentTimeMillis()) {
+            while (timeNext < SimpleTimeMark.now().toMillis()) {
                 timeNext += oneCircleTime
             }
-            val timeUntil = timeNext - System.currentTimeMillis()
+            val timeUntil = timeNext - SimpleTimeMark.now().toMillis()
             kingTime[king] = timeUntil
         }
         return kingTime
```

Comment:
Just some reformating of the code, but should work the same if I did not make a mistake (pls check that it is still correct).
```diff
-    private fun getKingTimes(): MutableMap<String, Long> {
+    private fun getKingTimes(): Map<String, Duration> {
```
```suggestion
        val sbTime = SkyBlockTime.now()
        val absDay = (sbTime.year - 1) * 372 + (sbTime.month - 1) * 31 + (sbTime.day - 1)
        val kingIndex = (absDay + 4) % kingCircles.size
        val circleStart = SkyBlockTime.fromAbsoluteDay(absDay - kingIndex)
        val oneSBDay = 20.minutes
        val oneCircleTime = oneSBDay * kingCircles.size
        
        val startTime = circleStart.toTimeMark().passedSince()
        
        return kingCircles.mapIndexed{ index, king ->
            val timeNext = startTime + oneSBDay * index
            king to if(timeNext.isNegative()) timeNext + oneCircleTime * (timeNext/ oneCircleTime + 1) else timeNext
        }.toMap()
    }
```
and in nextMissingText() the milliseconds call can go.

Reply:
added